wrangler config - add extended configuration docs (#18146)
cloudflare/cloudflare-docs: src/content/docs/workers/wrangler/configuration.mdx

If you change your routes in the dashboard, Wrangler will override them in the next deploy with the routes you have set in your `wrangler.toml`. To manage routes via the Cloudflare dashboard only, remove any route and routes keys from your `wrangler.toml` file. Then add `workers_dev = false` to your `wrangler.toml` file. For more information, refer to [Deprecations](/workers/wrangler/deprecations/#other-deprecated-behavior).

Wrangler will not delete your secrets (encrypted environment variables) unless you run `wrangler secret delete <key>`.

## Extending user configuration

> The main goal here is to allow tools to generate a partial configuration file that is merged into the user configuration when Wrangler commands are run.

Reviewer: ```suggestion
You can allow tools to generate a partial configuration file that is merged into the user configuration when Wrangler commands are run.
```

> The file must be written to `./.wrangler/config/extra.json`. This path is relative to the project path, which is the directory containing the wrangler.toml or the current working directory if there is no wrangler.toml.

Reviewer: ```suggestion
Write the partial configuration file to `./.wrangler/config/extra.json` in the current project directory, which is typically the directory containing the wrangler.toml.
```

> The format of the file is a JSON object whose properties are the [inheritable](#inheritable-keys) and [non-inheritable](#non-inheritable-keys) keys described above. Notably it cannot contain the [top level only](#top-level-only-keys) configuration keys.

Reviewer: ```suggestion
The file must be a JSON object, containing the [inheritable](#inheritable-keys) and [non-inheritable](#non-inheritable-keys) keys described above. It cannot contain [top level only](#top-level-only-keys) keys.
```

> The configuration in this file will be merged into the configuration of the currently selected environment before being used in all Wrangler commands.

Reviewer: ```suggestion
The configuration in this file is merged with the the currently selected environment and applied to all Wrangler commands. No manual action is required because Wrangler automatically detects and applies the file when it is present.
```

> 

Reviewer: ```suggestion
```

> The user does not need to manually specify that this merging should happen. It is done automatically when the file is found.

Reviewer: ```suggestion
```

Reviewer: Added this into the sentence above
